docs/specs/openapi.yaml

```
@@ -136,18 +339,227 @@ paths:
+                    }
+                  ]
+                }
+            multipart/form-data: # read
+              schema:
+                $ref: "#/components/schemas/ReadFilesResponse"
+              example:
+                files:
+                  "foo some file content"
+                response:
+                  {
+                    "type": "sync",
+                    "status-code": 200,
+                    "status": "OK",
+                    "result": [
+                      {
+                        "path": "/home/ubuntu/bar"
+                      }
+                    ]
+                  }
+    post:
+      summary: Create, write, remove files/directories.
+      tags:
+        - files
+      description: |
+        Write content to a path on the remote system;
+        
+        creates a directory or directory tree;
+
+        delete a file or directory.
+      requestBody:
+        content:
+          multipart/form-data:  # action: write
+            schema:
+              type: object
+              properties:
+                request:
+                  type: string
+                  description: |
+                    JSON metadata about the files to write.
+
+                    The format is binary because it's in a multipart part.
+
+                    Example: '{"action": "write", "files": [{"path": "/home/ubuntu/foo", "make-dirs": true, "permissions": "644"}]}'
+                  format: binary
+                files:
+                  type: array
+                  items:
+                    type: string
+                    format: binary
+                  description: |
+                    The files to be written.
+                    
+                    Each file is a separate part.
+                    
+                    For the file part, "Content-Type" is "application/octet-stream".
+
+                    "Content-Disposition" is "form-data; name="files"; filename=foo".
+          application/json:  # actin: make-dirs, remove
+            schema:
+              oneOf:  # Use oneOf since only one action can be performed at a time.
+                - $ref: "#/components/schemas/PostFilesMakeDirsRequest"
+                - $ref: "#/components/schemas/PostFilesRemovePathsRequest"
+            description: JSON payload for "make-dirs" or "remove" actions.
+      responses:
+        "200":
+          description: The result in the response is a JSON array of the file result object containing path and (optional) errors.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PostFilesResponse"
+              example:
+                {"type":"sync","status-code":200,"status":"OK","result":[{"path":"/home/ubuntu/foo"}]}
+  /v1/exec:
+    post:
+      summary: Execute a command
+      tags:
+        - files
+      description: Start a command with the given options, return a value representing the process.
```

Minor readability suggestion:
```suggestion
      description: Start a command with the given options and return a value representing the process.
```